File: Sorting Algorithms/Insertion_Sort.py
n = int(input("Enter the number of elements in the list: "))
arr = []
for i in range (n):
    arr.append(int(input("Enter the element: ")))
    
for i in range(1, n):
    curr = arr[i]
    prev = i-1
    
    while(prev >=0 and arr[prev] > curr):
        arr[prev + 1] = arr[prev]
        prev = prev-1
        
    arr[prev+1] = curr
    
print("The sorted list is: ")
for i in range(n):
    print(arr[i], end = " ")

File: Sorting Algorithms/Insertion_Sort.cpp
#include<iostream>
using namespace std;

int main() {
    int n;
    cout<<"Enter the number of elements: ";
    cin>>n;

    int arr[n];
    cout<<"Enter the elements: ";
    for(int i=0; i<n; i++) {
        cin>>arr[i];
    }

    for(int i=1;i<n;i++) {
        int curr = arr[i];
        int prev = i-1;
        
        while(prev >= 0 && arr[prev] > curr) {
            arr[prev+1] = arr[prev];
            prev--;
        }

        arr[prev+1] = curr;
    }

    cout<<"Sorted Array: ";
    for(int i=0;i<n;i++) {
        cout<<arr[i]<<" ";
    }

    return 0;
}